augur/filter.py

```diff
@@ -12,17 +12,21 @@
 
 comment_char = '#'
 
-def read_vcf(compressed, input_file):
-    import gzip
-    opn = gzip.open if compressed else open
 
-    with opn(input_file, mode='rt') as f: #'rt' necessary for gzip
```

Comment:
it is a pity to lose the context manager usage here.  Should at least put the `file.close()` in a finally block?

Reply:
Actually, I wonder if this makes sense. The more I think about it, the more I like it.

```python
def read_vcf(filename):
    with open_vcf(filename) as file:
        chrom_line = next(line for line in file if line.startswith("#C"))
        headers = chrom_line.strip().split("\t")
        sequences = headers[headers.index("FORMAT") + 1:]

        # because we need 'seqs to remove' for VCF
        return sequences, sequences.copy()


def open_vcf(filename):
    if filename.lower().endswith(".gz"):
        import gzip
        return gzip.open(filename, mode="rt")
    else:
        return open(filename)
```